MetaMask/metamask-extension
# feat(MMPD-554): Automatically switch to account initiating a transactions when a user is on a different account

## Explanation


Currently, if a dapp is connected to a different account than the one a user has selected, sending transactions provides a minimal amount of information. Only a single warning is shown. This is inconvenient for users since going back to the extension to check on the transaction, leaves it nowhere to be seen unless they switch to the account manually.

This now makes it so that the extension will automatically switch to the proper account, allowing users a faster flow and easier integration with dapps.

Link to Portfolio ticket (internal MM) https://consensyssoftware.atlassian.net/browse/MMPD-554



## Screenshots/Screencaps

![mmpd-554](https://github.com/MetaMask/metamask-extension/assets/22625602/d8918367-9f01-4768-b3e5-8583ad9e92fc)


## Manual Testing Steps

1. Connect account 1 to a dapp
1. switch to a different account in extension (ex: Account 2
1.start from the dapp any transaction such as transfer, spending approval or contract interaction
1. Approve transaction 
1. open extension, expect to see the different account now selected



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
